inbound_test.go

```diff
@@ -142,3 +142,38 @@ func TestInboundConnection_CallOptions(t *testing.T) {
 		require.NoError(t, err, "Call through proxy failed")
 	})
 }
+
+func TestBlackhole(t *testing.T) {
+	ctx, cancel := NewContext(testutils.Timeout(time.Hour))
+
+	testutils.WithTestServer(t, nil, func(server *testutils.TestServer) {
+		serviceName := server.ServiceName()
+		handlerName := "test-handler"
+
+		server.Register(HandlerFunc(func(ctx context.Context, inbound *InboundCall) {
+			// cancel client context in handler so the client can return after being blackholed
+			defer cancel()
+
+			c, _ := InboundConnection(inbound)
+			require.NotNil(t, c)
+
+			state := c.IntrospectState(&IntrospectionOptions{})
+			require.Equal(t, 1, state.InboundExchange.Count, "expected exactly one inbound exchange")
+
+			// blackhole request
+			inbound.Response().Blackhole()
+
+			// give time for exchange to cleanup
+			time.Sleep(10 * time.Millisecond)
+
+			state = c.IntrospectState(&IntrospectionOptions{})
+			require.Equal(t, 0, state.InboundExchange.Count, "expected no inbound exchanges")
+		}), handlerName)
+
+		clientCh := server.NewClient(nil)
+		defer clientCh.Close()
+
+		_, _, _, err := raw.Call(ctx, clientCh, server.HostPort(), serviceName, handlerName, nil, nil)
+		require.Error(t, err, "expected to timeout")
```

Comment:
I think you're actually expecting a cancelled error, since you call `cancel`.

Can you also add a `assert.Equal(t, ErrCodeCanceled, GetSystemErrorCode(err))` (or similar) to make sure we're getting an error for the right reason.